web/web/auth/static/actions.js

```diff
@@ -1,6 +1,95 @@
+import { api } from "../../static/js/network_client";
+import { menuRoutes } from "../../static/js/config/routes";
+import { createLoginError } from "./helpers";
+
 export function loginSuccessful(token) {
+    window.location.href = menuRoutes[0].path;
     return {
         type: "LOGIN_SUCCESSFUL",
-        token
-    }
-}
\ No newline at end of file
+        token,
+    };
+}
+
+export function loginFailed(token) {
+    return {
+        type: "LOGIN_FAILED",
+        token,
+    };
+}
+
+export function processLogin(username, password) {
+    return (dispatch) => {
+        const json = {
+            json: {
+                username: username,
+                password_hash: password,
+            },
+        };
+        api.post(
+            "validate_login",
+            json,
+            (data) => {
+                dispatch(loginSuccessful);
+            },
+            (error) => {
+                createLoginError(
+                    "Login failed",
+                    "Incorrect username and/or password"
+                );
+                dispatch(loginFailed);
+            }
+        );
+    };
+}
+
+export function processSignUp(username, firstName, lastName, password) {
+    return (dispatch) => {
+        try {
+            // Placeholders for address_id and utility_id
+            const userJson = {
+                json: {
+                    email: username,
+                    first_name: firstName,
+                    last_name: lastName,
+                    address_id: "1",
+                    utility_id: "1",
+                },
+            };
+            api.post(
+                "user",
+                userJson,
+                (data) => {
+                    const userId = data.results.data.id.toString();
+                    const loginJson = {
+                        json: {
+                            username: username,
+                            password_hash: password,
+                            user_id: userId,
+                        },
+                    };
+                    api.post(
```

Comment:
I think this request shouldn't be triggered from the frontend. After the `user` object creation on the backend, you should just go ahead and create the login as well. Leaving it up to frontend leads to integrity issues. If the user creation succeeds but the create login fails. The user will be in a weird state where they can't login.